jetkit/api/auth/__init__.py

```diff
@@ -82,15 +82,29 @@ def refresh_token():
         return {"access_token": create_access_token(identity=current_user)}
 
 
-def use_sign_up_api(auth_model: AuthModel, user_schema: Type[Schema] = UserSchema):
-    # Since sign up can require not only email/password, seperate this from core auth api
+def use_sign_up_api(
+        auth_model: AuthModel,
+        user_schema: Type[Schema] = UserSchema,
+        allowed_domains: Optional[List] = None,
+        allowed_emails: Optional[List] = None,
+):
+    # Since sign up can require not only email/password, separate this from core auth api
     @blp.route("sign-up", methods=["POST"])
     @blp.response(user_schema)
     @blp.arguments(AuthRequest, as_kwargs=True)
     def sign_up(email: str, password: str):
         """Sign up with email and password. Possibly add other fields later."""
         cleaned_email = email.strip().lower()
-        existing_user: AuthModel = auth_model.query.filter_by(email=cleaned_email).one_or_none()
+
+        domain = email[email.index('@') + 1:]
```

Comment:
what if `@` doesn't appear in the email?
parsing email addresses this way looks sketchy. you can use https://docs.python.org/3/library/email.utils.html#email.utils.parseaddr

Reply:
`AuthRequest` is a `marshmallow.Schema` that validates an email.
if the email does not contain `@`, an exception is thrown.
also, verified that each email contains exactly one `@`

`email.utils.parseaddr` does different job.
`utils.parseaddr('valerii <valerii@jetbridge.com>') == ('valerii', 'valerii@jetbridge.com')`
don't see how it helps there.

`utils.parseaddr('a') == ('', 'a') - it does not validate an email also.
`

Reply:
ok